docs/secret-template.md

```diff
@@ -1,56 +1,124 @@
-### Secret Template
+## SecretTemplate
 
-By default generated secrets have predefined set of keys. In a lot of cases `Secret` consumers expect specific set of keys within `data`. To control customization of keys within a secret, you can configure use secret template.
+As of v0.9.0+, the `SecretTemplate` is now avaliable under the `secretgen.carvel.dev` API group.
 
-Fields:
+Secrets are a common method of encapsulating and inputing sensitive data into an API via reference, or to a process via volume mounting.  In order to make compatible Secrets for these use cases, it often requires imperatively reading of other resources (including other secrets) and imperatively creating the compatible secret using these resources' information.  `SecretTemplate` aims to provide a declarative way of doing this process.
 
-- `type` (string; optional) Overrides secret resource type
-- `stringData` (map[string]string; optional) Overrides secret data. Values go through variable expansion with each type providing a set of variables that can be used. For example `postgresql-password: $(value)` for password type. See "Secret Template" section for each secret type. Available in v0.3.0+ (earlier available `data` key is removed).
+The CRD `SecretTemplate` provides a way of provides "input resources" (other resources on the API) and templating out a Secret using information found on these resources.  It will then pick up changes to these resources and update the templatde Secret as necessary.
```

Comment:
```suggestion
The CRD `SecretTemplate` provides a way of defining "input resources" (other Kubernetes Resources) and templating out a Secret using information found on these resources.  It will then pick up changes to these resources and update the templated Secret as necessary.
```

Reply:
Yep ^